internal/oc/release.go

```diff
@@ -343,19 +343,35 @@ func (r *release) Extract(log logrus.FieldLogger, releaseImage string, releaseIm
 	return path, err
 }
 
-func (r *release) GetReleaseBinaryPath(releaseImage string, cacheDir string) (workdir string, binary string, path string) {
-	workdir = filepath.Join(cacheDir, releaseImage)
+func (r *release) GetReleaseBinaryPath(releaseImage string, cacheDir string, ocpVersion string) (workdir string, binary string, path string, err error) {
 	binary = "openshift-baremetal-install"
+
+	// use the statically linked binary for 4.16 and up since our container is el8
+	// based and the baremetal binary for those versions is dynamically linked
+	// against el9 libaries
+	staticLinkingRequiredVersion := version.Must(version.NewVersion("4.16.0-ec.6"))
+	v, err := version.NewVersion(ocpVersion)
```

Comment:
why not to provide it as version object? Just to remove error handling from here? WDYT?

Reply:
Not sure it makes a difference. Then all the callers would have to do the same handling.